Nos/Controller/RefreshController.swift

```diff
@@ -0,0 +1,36 @@
+import Foundation
+
+/// Defines a common interface for refreshing.
+@MainActor protocol RefreshController: Observable {
+    /// Whether a refresh should begin or not.
+    var shouldRefresh: Bool { get }
+
+    /// The last time the view was refreshed.
+    var lastRefreshDate: Date? { get }
+
+    /// Update the state of `shouldRefresh` to the given value.
+    func setShouldRefresh(_: Bool)
```

Comment:
Why not using { get set } above?

Reply:
The short answer is that the compiler doesn't allow it.

We're using this protocol from `PagedNoteListView` which is a `UIViewRepresentable`. So then we have to make it `var refreshController: RefreshController` in `PagedNoteListView`. And then we have to add `mutating` to `updateUIView(_:context:)`. And then we have to mark it as `@MainActor`. And then `PagedNoteListView` doesn't conform to `UIViewRepresentable` any more. Maybe there's a way to work it out but I don't know how.